rest/route/subscription_routes.go

```diff
@@ -60,6 +62,13 @@ func (s *subscriptionPostHandler) ParseAndValidate(ctx context.Context, r *http.
 			}
 		}
 
+		if dbSubscription.Owner != u.Username() {
```

Comment:
This is technically vulnerable to someone mucking around in javascript and changing the owner. Should we change the upsert find to also include owner, that way the back end is safe from this

Reply:
I don't think it's actually vulnerable because we force the user in this context to auth using their API key. We do the same sort of thing for other routes that require an authenticated user

Reply:
So here's the scenario: I have the ID of a subscription you own. I load my subscription page, but change data stored in Javascript to reflect your ID, and change owner to my username. I then make changes, and submit. The code as written will allow me to change a subscription you created.

I think all we have to do is change the Upsert to also search based on owner in addition to id

Reply:
Fixed. Trying to change the owner will cause a duplicate key error which is probably fine